assimilate.csh

```diff
@@ -78,7 +78,17 @@ else if ($?PBS_NODEFILE) then
    setenv    NUMTASKS_PERNODE `grep $ANY_OLD_NODE $PBS_NODEFILE | wc -l`
    setenv  MPIEXEC_MPT_DEBUG 0
    setenv MP_DEBUG_NOTIMEOUT yes
-   setenv          LAUNCHCMD mpiexec_mpt
+# Debugging filter hanging/churning after writing the 'forecast' stage files.
+# 2020-3-09; we changed the TMPDIR defined in .login to solve a Matlab problem.
+# but I'm guessing that didn't affect the TMPDIR definition in here.
+# I've changed it back to /dev/shm/raeder.
+# Let's see what it is in a batch job.
+   echo "assimilate.csh: TMPDIR = $TMPDIR"
+# Also, check peak memory usage.
+   module load peak_memusage
+# See peak_memusage, below
+   # setenv          LAUNCHCMD mpiexec_mpt
+   setenv          LAUNCHCMD 'mpiexec_mpt peak_memusage.exe'
```

Comment:
The webpage you listed in the commit message also indicates:
"# yyyy-mm-dd Context: Cheyenne MPT peak_memusage job. 
# Variable MPI_SHEPHERD is set in this job in order to
# enable peak_memusage. Do not propagate it to other MPT
# jobs as it may cause significant slowdown or timeout.
# Contact the CISL Consulting Services Group if you have
# questions about this.
setenv MPI_SHEPHERD true"

Reply:
OK - that is NOT what the comment looked like when I entered it!  I was not SHOUTING!  :)

Reply:
The MPI_SHEPHARD only appears in the web page section about 
binaries compiled with MPT.  Am I right in thinking that filter falls under
the next section: Intel MPI-compiled binary?

Reply:
the default MPI is MPT, unless you've done something special to select the intel version of MPI. (just using the intel compiler does not select which mpi you compile with.  the default is MPT.) so you really won't want to run with MPI_SHEPHERD set unless you're doing a one-off test of memory usage.

Reply:
I'll take your word for it, but I just want to document for my future reference
that I'm using **mpi**f90 in mkmf.template.intel, which does not
mention MPT anywhere.  Then the web page refers separately to binaries 
built with MPT and MPI, but it counts on users knowing that MPT is the default MPI library which mpif90 uses instead of the intel mpi library.

Reply:
just so you know - almost every MPI program anywhere is compiled with something called "mpif90".  that is a wrapper around any fortran compiler and any MPI library.  it tells you nothing about what is actually being used.  on our systems, what gets used only depends on what modules you have loaded.  "module list" includes MPT by default but it's possible to swap that for openmpi, or intel IMPI ("impi" is what the intel mpi library is called).